### PR TITLE
[#2740] Fix checking for persistent connections.

### PR DESCRIPTION
**Pull request by adiroiban**
Problem
======

Empirical was checking that request headers are made so that connection is persistent but did not really check if new connections are made from same host.

Why we got this
============

No tests. 3rd party lib used for testing was not supporting persisted connections.

Changes
=======

update dependencies

fix bug in _DefinedRequestHandler.protocol_version which was set as server_version instead of protocol version.

update code to do a strict check for peer IP address to check persistent connections.
add tests for persistent connections by using a new 3rd part lib for testing.

add explicit check to see if thread was joined or join only did timeout.

How to test
=========

reviewers: @alibotean 

check that changes make sense

### CODE REVIEW

**Comment by adiroiban** [comment]
needs-review

**Comment by adiroiban** [comment]
needs-review thanks!

File: chevah/empirical/mockup.py [hunk 1]
```diff
@@ -33,12 +34,21 @@
...
 class _StoppableHTTPServer(BaseHTTPServer.HTTPServer):
     """
-    BaseHTTPServer but with a stoppable server_forever.
+    Single connection HTTP server designed to respond to HTTP requests in
+    functional tests..
     """
+    server_version = 'ChevahTesting/0.1'
+    stopped = False
+    # Current connection served by the server.
+    #active_connection = None
```
Comment (alibotean) [inline chevah/empirical/mockup.py]: Remove comment and commented code.

File: chevah/empirical/mockup.py [hunk 2]
```diff
@@ -33,12 +34,21 @@
 
 class _StoppableHTTPServer(BaseHTTPServer.HTTPServer):
     """
-    BaseHTTPServer but with a stoppable server_forever.
+    Single connection HTTP server designed to respond to HTTP requests in
+    functional tests..
```
Comment (alibotean) [inline chevah/empirical/mockup.py]: Double .

File: release-notes.rst [hunk 3]
```diff
@@ -2,6 +2,24 @@ Release notes for chevah.empirical
...
+0.34.1 - 10/03/2015
+-------------------
+
+* Fix HTTPServerContext close on Windows.
+
+
+0.34.0 - 10/03/2015
+-------------------
+
+* Add strict checking for persistent connection by.
```
Comment (alibotean) [inline release-notes.rst]: Incomplete release noted log entry.

**Comment by alibotean** [comment]
Just some minor comments.

changes-approved

File: chevah/empirical/mockup.py (hunk 1 above)
Comment (adiroiban) [inline chevah/empirical/mockup.py]: code  needs to be uncommented... as this is a class member and should be defined here...
left if while trying to replicate a strage bug which occured once on SLES

File: chevah/empirical/mockup.py (hunk 2 above)
Comment (adiroiban) [inline chevah/empirical/mockup.py]: done

File: release-notes.rst (hunk 3 above)
Comment (adiroiban) [inline release-notes.rst]: done

**Pull request closed without merging**